sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/RedundantConditionalAroundAssignment.cs

```diff
@@ -64,8 +61,13 @@ protected override void Initialize(SonarAnalysisContext context)
                     var expression1Assignment = assignment.Left?.RemoveParentheses();
                     var expression2Assignment = assignment.Right?.RemoveParentheses();
 
-                    if (AreMatchingExpressions(expression1Condition, expression2Condition, expression2Assignment, expression1Assignment) ||
-                        AreMatchingExpressions(expression1Condition, expression2Condition, expression1Assignment, expression2Assignment))
+                    if (!AreMatchingExpressions(expression1Condition, expression2Condition, expression2Assignment, expression1Assignment) &&
+                        !AreMatchingExpressions(expression1Condition, expression2Condition, expression1Assignment, expression2Assignment))
+                    {
+                        return;
+                    }
+
+                    if (!(c.SemanticModel.GetSymbolInfo(assignment.Left).Symbol is IPropertySymbol))

```

Comment:
This is the actual change to allow check and set for properties, the rest is mainly refactoring.